docs/content/customization/generic/encryption-at-rest.md

```diff
@@ -0,0 +1,60 @@
++++
+title = "Encryption At REST"
++++
+
+`encryptionAtRest` variable enables encrypting kubernetes resources at REST using provided encryption provider.
+When this variable is set, kuberntetes secrets and configmaps are encrypted before writing them at `etcd`.
+
+If the `encryptionAtRest` property is not specified, then
+the customization will be skipped. The secrets and configmaps will not be stored as encrypted in `etcd`.
+
+We support following encryption providers
+
+- aescbc
+- secretbox
+
+More information about encryption at REST: [Encrypting Confidential Data at Rest
+](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/)
+
+## Example
+
+To encrypt configmaps and secrets for using `aescbc` and `secretbox` encryption providers:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: <NAME>
+spec:
+  topology:
+    variables:
+      - name: clusterConfig
+        value:
+          encryptionAtRest:
+            providers:
+              - aescbc: {}
+              - secretbox: {}
```

Comment:
Only one of these will usually be used, right? Let's update the example to highlight that.

Reply:
I forgot to revert this after adding validation. thanks for spotting it.